components/monitoring/base/grafana-app.yaml

```diff
@@ -0,0 +1,148 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  namespace: appstudio-workload-monitoring
+  name: grafana-serviceaccount
+  annotations:
+    argocd.argoproj.io/compare-options: IgnoreExtraneous
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: appstudio-workload-monitoring
+  name: ocp-injected-certs
+  labels:
+    config.openshift.io/inject-trusted-cabundle: "true"
+---
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  namespace: appstudio-workload-monitoring
+  name: grafana-access
+  annotations:
+    service.beta.openshift.io/serving-cert-secret-name: grafana-tls
+    kubernetes.io/tls-acme: "true"
+spec:
+  port:
+    targetPort: https
+  tls:
+    termination: reencrypt
+  to:
+    kind: Service
+    name: grafana-service
+    weight: 100
+  wildcardPolicy: None
+---
+apiVersion: integreatly.org/v1alpha1
+kind: Grafana
+metadata:
+  namespace: appstudio-workload-monitoring
+  name: grafana
+spec:
+  config:
+    log:
+      mode: "console"
+      level: "warn"
+    # auth.anonymous:
+    #   enabled: True
+    auth:
+      disable_login_form: False
+      disable_signout_menu: True
+    auth.basic:
+      enabled: False
+    auth.proxy:
+      enabled: True
+      enable_login_token: True
+      header_property: username
+      header_name: X-Forwarded-User
+    server:
+      http_port: "3001"
+      protocol: http
+  dashboardLabelSelector:
+    - matchExpressions:
+        - {key: app, operator: In, values: ["<label_to_search>"]}
+  containers:
+    - name: oauth2-proxy
+      image: quay.io/oauth2-proxy/oauth2-proxy:v7.3.0-amd64
+      imagePullPolicy: Always
+      args:
+      - --provider=github
+      - --https-address=:3000
+      - --email-domain=*
+      - --upstream=http://localhost:3001
+      - --tls-cert-file=/etc/tls/private/tls.crt
+      - --tls-key-file=/etc/tls/private/tls.key
+      - --skip-auth-regex=^/metrics
+      - --github-org=redhat-appstudio-sre
+      - --github-team=stage
+      env:
+      - name: OAUTH2_PROXY_CLIENT_ID
+        valueFrom:
+          secretKeyRef:
+            key: client-id
+            name: grafana-oauth2-proxy-config
+      - name: OAUTH2_PROXY_CLIENT_SECRET
+        valueFrom:
+          secretKeyRef:
+            key: client-secret
+            name: grafana-oauth2-proxy-config
+      - name: OAUTH2_PROXY_COOKIE_SECRET
+        valueFrom:
+          secretKeyRef:
+            key: cookie-secret
+            name: grafana-oauth2-proxy-config
```

Comment:
Let's make sure we are adding this secret to README as well

Reply:
this should be covered by https://github.com/redhat-appstudio/infra-deployments/pull/883